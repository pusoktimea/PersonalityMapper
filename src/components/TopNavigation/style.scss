@import '~compass-mixins/lib/compass';
@import "../../styles/_colors.scss";

$baseClass: '.activate-top-navigation';

#{$baseClass} {
  background-color: $mirage;
  z-index: 904;
  height: 66px;
  padding: 0 32px 0 40px;
  position: fixed;
  width: 100%;
  min-width: 320px;
  &_logo {
    color: $white;
    font-size: 24px;
    line-height: 60px;
    text-decoration: none;
    font-family: Roboto,sans-serif;
    white-space: nowrap;
    float: left;
    outline: 0!important;
    display: block;
    &:hover {
      color: $jungleGreen;
    }
    span {
      color: $jungleGreen;
    }
  }
  &_burger {
    color: $white;
    font-size: 30px;
    margin: 15px;
    &:hover {
      color: $jungleGreen;
      cursor: pointer;
    }
  }
  &_profile {
    float: right;
    min-width: 230px;
    margin-top: 10px;
    margin-right: 75px;
    &_name {
      text-decoration: none;
      color: $white;
      font-size: 16px;
      line-height: 45px;
      font-family: Roboto,sans-serif;
      white-space: nowrap;
      float: left;
      outline: 0!important;
      display: block;
      &:hover {
        color: $jungleGreen;
      }
      span {
        color: $jungleGreen;
      }
    }
    &_image {
      background: $scooter;
      width: 45px;
      height: 45px;
      @include border-radius(50%);
      float: right;
    }
  }
}
.profile-dropdown {
  padding: 0;
  margin: 0;
  list-style: none;
  a {
    text-decoration: none;
    text-align: left;
    border: none;
    text-decoration: none;
    color: #7d7d7d;
    padding: 4px 16px 4px 20px;
    display: block;
    clear: both;
    font-weight: 400;
    line-height: 1.42857143;
    white-space: nowrap;
    cursor: pointer;

    .activate-icon {
      margin-right: 10px;
    }
  }
}
.profile-dropdown_item:hover {
  background-color: $jungleGreen;
  a {
    color: $white;
  }
}
.rc-tooltip {
  left: auto !important;
  right: 22px;
}
.rc-tooltip-inner {
  float: left;
  min-width: 160px;
  padding: 5px 0;
  margin: 2px 0 0;
  font-size: 14px;
  text-align: left;
  list-style: none;
  background-color: $white !important;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  border: 1px solid $dustyGray;
  border: 1px solid rgba(0,0,0,.15);
  @include border-radius(4px);
  @include box-shadow(rgba(0,0,0,.175) 0 6px 12px);
}
.rc-tooltip-placement-bottomRight .rc-tooltip-arrow {
    top: -10px;
    display: block;
    width: 0;
    height: 0;
    border: 11px solid transparent;
    border-bottom-color: $white !important;
}
