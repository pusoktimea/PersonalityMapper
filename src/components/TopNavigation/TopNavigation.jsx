import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

import {doLogout} from 'utils/LoginAPIUtils';

import Tooltip from '../Tooltip';
import Icon from '../Icon';

import img from './Boss-3.svg';

import './style.scss';

class TopNavigation extends PureComponent {
  static propTypes = {
    history: PropTypes.object.isRequired,
    profile: PropTypes.object.isRequired,
    isProfileLoaded: PropTypes.bool.isRequired,
    onGetProfile: PropTypes.func.isRequired,
    isSideBarMinimised: PropTypes.bool,
    onSideBarToggle: PropTypes.func
  }

  componentDidMount() {
    if (!this.props.isProfileLoaded) {
      this.props.onGetProfile();
    }
  }

  logoutHandler = () => {
    doLogout().then(() => {
      this.props.history.push('/login');
    });
  }

  render() {
    const {profile} = this.props;
    return (
      <div className="activate-top-navigation">
        <Link to="/dashboard" className="activate-top-navigation_logo"><span>Activate</span>Admin</Link>
        <Icon
          className="activate-top-navigation_burger"
          icon="bars"
          onClick={this.props.onSideBarToggle}
        />
        <div className="activate-top-navigation_profile">
          <Link to="/profile" className="activate-top-navigation_profile_name"><span>Hi, </span>{profile.full_name}</Link>
          <Tooltip
            overlay={(
              <ul className="profile-dropdown">
                <li className="profile-dropdown_item">
                  <Link to="/profile"><Icon icon="user" />Profile</Link>
                </li>
                <li className="profile-dropdown_item">
                  <a onClick={this.logoutHandler}><Icon icon="power-off" />Sign out</a>
                </li>
              </ul>)}
            placement="topRight"
            trigger="click">
            <img className="activate-top-navigation_profile_image" src={img} />
          </Tooltip>
        </div>
      </div>
    );
  }
}

export default TopNavigation;
