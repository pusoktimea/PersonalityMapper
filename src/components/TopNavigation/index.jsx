import React from 'react';
import {Link} from 'react-router-dom';
import img from './Boss-3.svg';
import Tooltip from '../Tooltip';
import Icon from '../Icon';
import logo from './SoftPsychology Consulting-2.png';

import './style.scss';

const TopNavigation = () => (
  <div className="persmap-top-navigation">
    <Link to="/dashboard" className="persmap-top-navigation_logo"><img src={logo} width="300" /></Link>
    <div className="persmap-top-navigation_profile">
      <Link to="/profile" className="persmap-top-navigation_profile_name"><span>Hi, </span>Timea Pusok</Link>
      <Tooltip
        overlay={(
          <ul className="profile-dropdown">
            <li className="profile-dropdown_item">
              <Link to="/profile"><Icon icon="user" />Profile</Link>
            </li>
            <li className="profile-dropdown_item">
              <Link to="/login"><Icon icon="power-off" />Sign out</Link>
            </li>
          </ul>)}
        placement="topRight"
        trigger="click">
        <img className="persmap-top-navigation_profile_image" src={img} />
      </Tooltip>
    </div>
  </div>
);

export default TopNavigation;
