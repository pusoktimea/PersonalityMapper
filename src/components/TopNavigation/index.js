import {connect} from 'react-redux';

import {doGetProfile, toggleSideBar} from 'actions/page';

import TopNavigation from './TopNavigation';

const mapStateToProps = (state) => {
  return {
    profile: state.page.profile,
    isProfileLoaded: state.page.isProfileLoaded,
    isSideBarMinimised: state.page.isSideBarMinimised
  };
};

const mapDispatchToProps = (dispatch) => ({
  onGetProfile: () => {
    dispatch(doGetProfile());
  },
  onSideBarToggle: () => {
    dispatch(toggleSideBar());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(TopNavigation);
