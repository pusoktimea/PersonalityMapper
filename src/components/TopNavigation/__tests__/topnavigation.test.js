import 'raf/polyfill';

import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import TopNavigation from '../TopNavigation';

describe('Top Navigation test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <Router>
          <TopNavigation
            history={{}}
            profile={{full_name: 'foo bar'}}
            isProfileLoaded
            onGetProfile={jest.fn()}
          />
        </Router>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
