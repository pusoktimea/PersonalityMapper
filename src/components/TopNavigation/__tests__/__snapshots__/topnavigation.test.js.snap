// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Navigation test renders and matches snapshot 1`] = `
<div
  className="activate-top-navigation"
>
  <a
    className="activate-top-navigation_logo"
    href="/dashboard"
    onClick={[Function]}
  >
    <span>
      Activate
    </span>
    Admin
  </a>
  <span
    className="activate-icon fa fa-bars activate-top-navigation_burger"
    id={undefined}
    onClick={undefined}
  />
  <div
    className="activate-top-navigation_profile"
  >
    <a
      className="activate-top-navigation_profile_name"
      href="/profile"
      onClick={[Function]}
    >
      <span>
        Hi, 
      </span>
      foo bar
    </a>
    <img
      className="activate-top-navigation_profile_image"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      src={Object {}}
    />
  </div>
</div>
`;
